plugins/inputs/snmp/snmp.go

```diff
@@ -476,14 +472,17 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 				rtr.Fields = map[string]interface{}{}
 				rows[i] = rtr
 			}
-			if f.IsTag {
-				if vs, ok := v.(string); ok {
-					rtr.Tags[f.Name] = vs
+			// don't add an empty string
+			if vs, ok := v.(string); !ok || vs != "" {
```

Comment:
If telegraf core automatically removes empty tags before passing to influxdb (as sparrc indicated in https://github.com/influxdata/telegraf/pull/1927#issuecomment-255739321), I think we should let telegraf core do so instead of explicitly doing it here.